zerver/views/messages.py

```diff
@@ -829,16 +816,30 @@ def mark_all_as_read(request, user_profile):
     return json_success({'result': 'success',
                          'msg': ''})
 
-def update_stream_topic_message_flags(request,
-                                      user_profile,
-                                      operation,
-                                      flag,
-                                      stream_name,
-                                      topic_name):
-    # type: (HttpRequest, UserProfile, Text, Text, Optional[Text], Optional[Text]) -> HttpResponse
-    assert(operation == 'add')
-    assert(flag == 'read')
+@has_request_variables
+def mark_stream_as_read(request,
+                        user_profile,
+                        stream_name=REQ()):
+    # type: (HttpRequest, UserProfile, Text) -> HttpResponse
+    try:
+        stream = get_stream(stream_name, user_profile.realm)
+    except Stream.DoesNotExist:
+        raise JsonableError(_('No such stream \'%s\'') % (stream_name,))
+
+    count = do_mark_stream_messages_as_read(user_profile, stream)
+
+    log_data_str = "[mark_stream_as_read/%s updated]" % (count,)
+    request._log_data["extra"] = log_data_str
```

Comment:
I'm going to fix this while merging since that seems faster, but FYI, `extra_log_data` gets included in the log lines in the `run-dev.py` console output and `server.log` on the server.  These log lines always contain the endpoint by default; `extra_log_data` is to provide extra details that can help in understanding if something is a problem (e.g. if marking a stream as read was slow because there were 5000 unread messages, that has an explanation!).  So we only need the `"%s updated"` part.